### ISSUE TITLE
Small scenery object count is off

### ISSUE THREAD

**Issue by Liampie** [OpenRCT2/OpenRCT2#8350]
So in the object selection window it says I have 251 small scenery objects in use, but when I select a 252th object, and close the object selection window, the new object is nowhere to be found. Going back to the object selection window shows 251 objects in use, and the object I selected being unselected.

Opening the savegame in parkdat proves that actually all 252 slots are filled, and therefore the count must be wrong.

Here's the savegame in question:
https://www.nedesigns.com/park/4290/seaquarium-curacao/download/

**Comment by jmyre1999** [comment]
My team and I would like to work on this issue

**Comment by Nazey** [comment]
It seems like something is inherently off by 1 in the actual base game for small scenery objects. Everything else in the developer options seems to stick to their maximum values, but small scenery refuses to. Is it possible to debug the base game with the trial of the game, or do I need to buy a copy?

**Comment by IntelOrca** [comment]
@Nazey you will need to buy Rollercoaster Tycoon 2 in order to run or debug OpenRCT2. Since you are working together as a group I don't see harm in buying one copy and then sharing the game files. You can get RCT2 cheap and even cheaper during the sales.

https://isthereanydeal.com/game/rollercoastertycooniitriplethrillpack/info/

**Comment by Nazey** [comment]
@IntelOrca Yeah, I'm just gonna buy it. It's already cheap enough. How do I actually debug the base game? (I'm on osx, but if it's easier I have a Windows environment)

**Comment by duncanspumpkin** [comment]
I use ollydbg on Windows. Not sure what everyone else uses.

**Comment by IntelOrca** [comment]
If you want to debug the base game, it is better to use Windows. But I don't think you need to debug the vanilla binary to diagnose this issue. You just need to debug OpenRCT2 which can be done using xcode on macOS.

**Comment by Nazey** [comment]
I cannot for the life of me figure out where the count/variable for small scenery objects is. Tried debugging and watching for it, but to no avail. It is specific to this map only, however. I tested it with other maps where the count is correct and stays at 252.

**Comment by IntelOrca** [comment]
@Nazey `reset_selected_object_count_and_size` is what is calculating the number that is shown at the bottom of the window.

**Comment by spacek531** [comment]
@Liampie is this issue still happening in NSF?
